google-beta/resource_pubsub_subscription.go

```diff
@@ -1,228 +1,369 @@
+// ----------------------------------------------------------------------------
+//
+//     ***     AUTO GENERATED CODE    ***    AUTO GENERATED CODE     ***
+//
+// ----------------------------------------------------------------------------
+//
+//     This file is automatically generated by Magic Modules and manual
+//     changes will be clobbered when the file is regenerated.
+//
+//     Please read more about how to change this file in
+//     .github/CONTRIBUTING.md.
+//
+// ----------------------------------------------------------------------------
+
 package google
 
 import (
 	"fmt"
+	"log"
+	"reflect"
+	"regexp"
+	"strconv"
+	"time"
 
 	"github.com/hashicorp/terraform/helper/schema"
-	"google.golang.org/api/pubsub/v1"
-	"regexp"
 )
 
+func getComputedTopicName(project, topic string) string {
+	match, _ := regexp.MatchString("projects\\/.*\\/topics\\/.*", topic)
+	if match {
+		return topic
+	}
+	return fmt.Sprintf("projects/%s/topics/%s", project, topic)
+}
+
+func getComputedSubscriptionName(project, subscription string) string {
+	match, _ := regexp.MatchString("projects\\/.*\\/subscriptions\\/.*", subscription)
+	if match {
+		return subscription
+	}
+	return fmt.Sprintf("projects/%s/subscriptions/%s", project, subscription)
+}
+
 func resourcePubsubSubscription() *schema.Resource {
 	return &schema.Resource{
 		Create: resourcePubsubSubscriptionCreate,
 		Read:   resourcePubsubSubscriptionRead,
-		Update: resourcePubsubSubscriptionUpdate,
 		Delete: resourcePubsubSubscriptionDelete,
 
 		Importer: &schema.ResourceImporter{
-			State: resourcePubsubSubscriptionStateImporter,
+			State: resourcePubsubSubscriptionImport,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(240 * time.Second),
+			Delete: schema.DefaultTimeout(240 * time.Second),
 		},
 
 		Schema: map[string]*schema.Schema{
 			"name": {
-				Type:     schema.TypeString,
-				Required: true,
-				ForceNew: true,
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				DiffSuppressFunc: compareSelfLinkOrResourceName,
 			},
-
 			"topic": {
 				Type:             schema.TypeString,
 				Required:         true,
 				ForceNew:         true,
 				DiffSuppressFunc: compareSelfLinkOrResourceName,
 			},
-
 			"ack_deadline_seconds": {
 				Type:     schema.TypeInt,
-				Optional: true,
 				Computed: true,
+				Optional: true,
 				ForceNew: true,
 			},
-
-			"project": {
-				Type:     schema.TypeString,
+			"labels": {
+				Type:     schema.TypeMap,
 				Optional: true,
-				Computed: true,
 				ForceNew: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
 			},
-
-			"path": {
```

Comment:
In theory this is a breaking change (not captured by acceptance tests) as we removed `path`. But `path` was not an API member and it appears to be interchangeable for `id`.

Reply:
Ah yeah, I commented about it in my review (I didn't see this till after I was done)

Unfortunately we need to adhere to a _very very_ noisy versioning policy where removing fields like this would require a major version jump; sadly, we'll have to keep it until at least `3.0.0`.

Feel free to add a `TODO` to consider deprecating it, although I'd prefer keeping it non-deprecated for now as we don't want users to use `id`.